### PR TITLE
Apply CSP to all HTML pages

### LINKED ISSUE #2185: Apply CSP to dartdoc pages

### PR DESCRIPTION
**Pull request by isoos**
- fixes #2185
- Moved CSP header extension to a top-level handler wrapper.
- removed hostname from dartdoc customization (we required it while we were serving out of dartdocs.org)
- CSP needed an extra host, because older dartdoc content refers to the github style with full URL.

### CODE REVIEW

File: app/lib/shared/handler_helpers.dart [hunk 1]
```diff
@@ -33,6 +33,7 @@ Future<void> runHandler(
   }
   handler = _redirectToHttpsWrapper(handler);
   handler = _logRequestWrapper(logger, handler);
+  handler = _cspHeaderWrapper(handler);
```
Comment (jonasfj) [inline app/lib/shared/handler_helpers.dart]: Isn't this the same as self? Well, maybe not when we're on new domain :)

File: app/lib/shared/handler_helpers.dart [hunk 2]
```diff
@@ -49,6 +50,24 @@ Future<void> runHandler(
...
+shelf.Handler _cspHeaderWrapper(shelf.Handler handler) {
+  return (shelf.Request request) async {
+    final rs = await handler(request);
+    final contentType = rs.headers['content-type'];
+    final isHtml = contentType != null && contentType.startsWith('text/html');
+    if (isHtml) {
+      return rs.change(headers: {
+        'x-content-type-options': 'nosniff',
+        'content-security-policy': contentSecurityPolicy,
+      });
```
Comment (jonasfj) [inline app/lib/shared/handler_helpers.dart]: Don't we also need something about frame options to protect against click jacking in old browser..
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options

Comment (isoos) [reply app/lib/shared/handler_helpers.dart]: `X-Frame-Options: sameorigin`?

File: app/lib/shared/handler_helpers.dart (hunk 1 above)
Comment (isoos) [reply app/lib/shared/handler_helpers.dart]: The issue is with absolute URLs, and yeah, we may need to add an extra rule if we move to a new domain. But this PR already reduces the absolute URLs is dartdoc content, and hopefully the remaining ones are not heavily used.

File: app/lib/shared/handler_helpers.dart (hunk 2 above)
Comment (jonasfj) [reply app/lib/shared/handler_helpers.dart]: do we use frames anywhere? If not we can just forbid it, and allow it when we need to...

File: app/lib/shared/handler_helpers.dart (hunk 1 above)
Comment (jonasfj) [reply app/lib/shared/handler_helpers.dart]: ++

File: app/lib/shared/handler_helpers.dart (hunk 2 above)
Comment (isoos) [reply app/lib/shared/handler_helpers.dart]: ack, `deny` it is.

**Pull request merged**